src/docker/restorableappengine.cc

```diff
@@ -761,11 +781,18 @@ bool RestorableAppEngine::checkAppContainers(const App& app, const std::string&
 
 void RestorableAppEngine::pullImage(const std::string& client, const std::string& src,
                                     const boost::filesystem::path& dst_dir,
-                                    const boost::filesystem::path& shared_blob_dir, const std::string& format) {
+                                    const boost::filesystem::path& shared_blob_dir, int max_parallel_pulls,
+                                    const std::string& format) {
   boost::filesystem::create_directories(dst_dir);
-  exec(boost::format{"%s copy -f %s --dest-shared-blob-dir %s %s oci:%s"} % client % format % shared_blob_dir.string() %
-           src % dst_dir.string(),
-       "failed to pull image", boost::this_process::environment());
+  if (-1 == max_parallel_pulls) {
```

Comment:
should we just enforce your actual range and then not need to keep this if/else logic?

Reply:
Only LmP has the `skopeo` version that supports `--max-parallel-pulls` param, so we cannot run `skopeo copy` unconditionally. (e.g. running unit tests in the container or running aklite on a local host).